server/route.go

```diff
@@ -491,6 +484,24 @@ func routeSid(sub *subscription) string {
 	return fmt.Sprintf("%s%s:%d:%s", qi, RSID, sub.client.cid, sub.sid)
 }
 
+func parseRouteSid(rsid []byte) (matches [EXPECTED_MATCHES][]byte, ok bool) {
```

Comment:
Since you get rid of regex, why don't return `[]byte, []byte, bool` instead? Why carry the array of matches that is specific to regex?